File: exampleSources/java/app2.java
// Imperative loop filtering tasks by status.

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

public final class Task {
    public enum Status { OPEN, DONE, BLOCKED }

    public final String title;
    public final Status status;
    public final int priority;

    public Task(String title, Status status, int priority) {
        this.title = title;
        this.status = status;
        this.priority = priority;
    }

    public boolean isOpen() {
        return status == Status.OPEN;
    }
}

class Main {
    public static void main(String[] args) {
        List<Task> backlog = new ArrayList<>();
        backlog.add(new Task("Write tests", Task.Status.OPEN, 2));
        backlog.add(new Task("Fix login bug", Task.Status.BLOCKED, 5));
        backlog.add(new Task("Refactor parser", Task.Status.DONE, 3));
        backlog.add(new Task("Ship release", Task.Status.OPEN, 1));

        List<Task> open = new ArrayList<>();
        for (Task t : backlog) {
            if (t.isOpen()) {
                open.add(t);
            }
        }
        Collections.sort(open, Comparator.comparingInt(t -> t.priority));

        for (Task t : open) {
            System.out.println(t.title);
        }
    }
}

File: exampleSources/csharp/app2.cs
// C# imperative foreach filtering tasks by status.

using System;
using System.Collections.Generic;

public enum Status { Open, Done, Blocked }

public sealed class Task
{
    public string Title;
    public Status StatusValue;
    public int Priority;

    public Task(string title, Status status, int priority)
    {
        Title = title;
        StatusValue = status;
        Priority = priority;
    }

    public bool IsOpen() => StatusValue == Status.Open;
}

public class Program
{
    public static void Main()
    {
        var backlog = new List<Task>
        {
            new Task("Write tests", Status.Open, 2),
            new Task("Fix login bug", Status.Blocked, 5),
            new Task("Refactor parser", Status.Done, 3),
            new Task("Ship release", Status.Open, 1),
        };

        var open = new List<Task>();
        foreach (var t in backlog)
        {
            if (t.IsOpen())
            {
                open.Add(t);
            }
        }
        open.Sort((a, b) => a.Priority.CompareTo(b.Priority));

        foreach (var t in open)
        {
            Console.WriteLine(t.Title);
        }
    }
}